### PR TITLE
api: redefines Txn API and hides Txn Grpc Response

### PR DESCRIPTION
**Pull request by fanminshi**

### CODE REVIEW

File: src/main/java/com/coreos/jetcd/Txn.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+ */
+public interface Txn {
+
+  /**
+   * takes a list of comparison. If all comparisons passed in succeed,
+   * the operations passed into Then() will be executed. Or the operations
+   * passed into Else() will be executed.
+   */
+  //CHECKSTYLE:OFF
+  Txn If(Cmp... cmps);
```
Comment (xiang90) [inline src/main/java/com/coreos/jetcd/Txn.java]: if ?

File: src/main/java/com/coreos/jetcd/Txn.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  //CHECKSTYLE:ON
+
+  /**
+   * takes a list of operations. The Ops list will be executed, if the
+   * comparisons passed in If() succeed.
+   */
+  //CHECKSTYLE:OFF
+  Txn Then(Op... ops);
```
Comment (xiang90) [inline src/main/java/com/coreos/jetcd/Txn.java]: then?

File: src/main/java/com/coreos/jetcd/Txn.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  //CHECKSTYLE:ON
+
+  /**
+   * takes a list of operations. The Ops list will be executed, if the
+   * comparisons passed in If() fail.
+   */
+  //CHECKSTYLE:OFF
+  Txn Else(Op... ops);
```
Comment (xiang90) [inline src/main/java/com/coreos/jetcd/Txn.java]: else?

File: src/main/java/com/coreos/jetcd/kv/TxnResponse.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
+package com.coreos.jetcd.kv;
+
+import com.coreos.jetcd.data.Header;
+import java.util.List;
+
+public class TxnResponse {
```
Comment (xiang90) [inline src/main/java/com/coreos/jetcd/kv/TxnResponse.java]: doc string?

File: src/main/java/com/coreos/jetcd/kv/TxnResponse.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    this.deleteResponses = deleteResponses;
+    this.getResponses = getResponses;
+    this.putResponses = putResponses;
+  }
+
+  public Header getHeader() {
+    return header;
+  }
+
+  public List<DeleteResponse> getDeleteResponses() {
```
Comment (xiang90) [inline src/main/java/com/coreos/jetcd/kv/TxnResponse.java]: doc string?

File: src/main/java/com/coreos/jetcd/kv/TxnResponse.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    return deleteResponses;
+  }
+
+  public List<GetResponse> getGetResponses() {
```
Comment (xiang90) [inline src/main/java/com/coreos/jetcd/kv/TxnResponse.java]: doc string?

File: src/main/java/com/coreos/jetcd/kv/TxnResponse.java [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    return getResponses;
+  }
+
+  public List<PutResponse> getPutResponses() {
```
Comment (xiang90) [inline src/main/java/com/coreos/jetcd/kv/TxnResponse.java]: doc string?

**Comment by xiang90** [comment]
lgtm

File: src/main/java/com/coreos/jetcd/Txn.java (hunk 1 above)
Comment (fanminshi) [inline src/main/java/com/coreos/jetcd/Txn.java]: `if` is a reserved key word in java for if statement.  can't use that for function name.

File: src/main/java/com/coreos/jetcd/Txn.java (hunk 3 above)
Comment (fanminshi) [inline src/main/java/com/coreos/jetcd/Txn.java]: `else` is also reserved. can't use that.

File: src/main/java/com/coreos/jetcd/kv/TxnResponse.java (hunk 5 above)
Comment (fanminshi) [inline src/main/java/com/coreos/jetcd/kv/TxnResponse.java]: will do

File: src/main/java/com/coreos/jetcd/kv/TxnResponse.java (hunk 4 above)
Comment (fanminshi) [inline src/main/java/com/coreos/jetcd/kv/TxnResponse.java]: will do.

File: src/main/java/com/coreos/jetcd/kv/TxnResponse.java (hunk 6 above)
Comment (fanminshi) [inline src/main/java/com/coreos/jetcd/kv/TxnResponse.java]: will do

File: src/main/java/com/coreos/jetcd/Txn.java (hunk 1 above)
Comment (xiang90) [inline src/main/java/com/coreos/jetcd/Txn.java]: oh... i see. keep if :P

File: src/main/java/com/coreos/jetcd/kv/TxnResponse.java (hunk 7 above)
Comment (fanminshi) [inline src/main/java/com/coreos/jetcd/kv/TxnResponse.java]: will do

File: src/main/java/com/coreos/jetcd/Txn.java (hunk 3 above)
Comment (xiang90) [inline src/main/java/com/coreos/jetcd/Txn.java]: ack

**Pull request merged**